src/materials/czm/CZMMaterialBasePK1.C

```diff
@@ -142,6 +169,11 @@ void CZMMaterialBasePK1::computeF() {
                                             (*_grad_disp_neighbor[2])[_qp]));
 
   _F_avg[_qp] = 0.5 * (F + F_neighbor);
+  for (uint i = 0; i < 3; i++)
+    for (uint j = 0; j < 3; j++)
+      if (!std::isfinite(_F_avg[_qp](i, j)))
```

Comment:
This seems to happen even in production mode.  Is that what you intended/does it slow things down considerably?

Reply:
Yes, this is where it is supposed to be. Doesn't seem to affect speed but I didn't try in a serious simulation.
the other option is to try to catch the mooseError, generated by the RankTwoTensorRU decomposition, that should be much faster... let me check

Reply:
according to Cody mooseError are always fatal, so nothing we can do about it.
The norm of the tensor might work, but there is the risk of an unwanted overflow (I tried and it happens).
SO checking component by component remains the only reasonable strategy